.github/workflows/ci.yml

```diff
@@ -453,6 +453,8 @@ jobs:
         if: always()
       - run: script/test/test2annotation.sh *-gotest.json
         if: always()
+      - run: sudo dmesg -T -f kern
```

Comment:
```plain
2023-05-25T05:31:07.6686761Z [Thu May 25 05:29:39 2023] EXT4-fs (loop8): mounted filesystem with ordered data mode. Opts: (null). Quota mode: none.
2023-05-25T05:31:07.6687461Z [Thu May 25 05:29:39 2023] ext4 filesystem being mounted at /tmp/snapshot-suite-Blockfile-3001051953/work/nextlayer supports timestamps until 2038 (0x7fffffff)
2023-05-25T05:31:07.6687932Z [Thu May 25 05:29:39 2023] loop10: detected capacity change from 0 to 262144
2023-05-25T05:31:07.6688425Z [Thu May 25 05:29:39 2023] EXT4-fs (loop10): INFO: recovery required on readonly filesystem
2023-05-25T05:31:07.6688980Z [Thu May 25 05:29:39 2023] EXT4-fs (loop10): write access unavailable, cannot proceed (try mounting with noload)
2023-05-25T05:31:07.6689428Z [Thu May 25 05:29:39 2023] EXT4-fs (loop9): recovery complete
```

Reply:
I think it maybe caused by DETACH umount. 
Could we umount the snapshot without DETACH?
ping @dmcgowan 


REF: https://github.com/containerd/containerd/blob/b16b0c872d3402d8e51499a30393f444f0f8f0e7/pkg/testutil/mount_linux.go#L21

Reply:
In my local devbox, it looks the same about duration. Ideally, the umount will sync the data?

Reply:
`MNT_DETACH` should allow the umount(2) to succeed immediately but the actual unmount only complete once the data has been synced.

Reply:
I run the following change in my devbox and run into this error.

```
        testsuite.go:940: mount /tmp/snapshot-suite-Blockfile-3062786410/work/prepare-layer-71
        helpers.go:88: unmount /tmp/snapshot-suite-Blockfile-3062786410/work/prepare-layer-71
        testsuite.go:940: mount /tmp/snapshot-suite-Blockfile-3062786410/work/prepare-layer-72
        helpers.go:88: unmount /tmp/snapshot-suite-Blockfile-3062786410/work/prepare-layer-72
        testsuite.go:944: [layer 72] preparing doesn't equal to flat before apply: directory diff between /tmp/snapshot-suite-Blockfile-3062786410/work/flat and /tmp/snapshot-suite-Blockfile-3062786410/work/prepare-layer-72
            ~ /onlyme/file-71(mode: 777, uid: 0, gid: 0) -> /onlyme/file-71(mode: 755, uid: 0, gid: 0)
        helpers.go:69: drwx------       4096 /tmp/snapshot-suite-Blockfile-3062786410
```

```diff
diff --git a/pkg/testutil/mount_linux.go b/pkg/testutil/mount_linux.go
index 9c4cf8c16..908d9ad00 100644
--- a/pkg/testutil/mount_linux.go
+++ b/pkg/testutil/mount_linux.go
@@ -16,6 +16,4 @@

 package testutil

-import "golang.org/x/sys/unix"
-
-const umountflags int = unix.MNT_DETACH
+const umountflags int = 0 // unix.MNT_DETACH
diff --git a/snapshots/blockfile/blockfile_loopsetup_test.go b/snapshots/blockfile/blockfile_loopsetup_test.go
index 06b4dd306..bf263478d 100644
--- a/snapshots/blockfile/blockfile_loopsetup_test.go
+++ b/snapshots/blockfile/blockfile_loopsetup_test.go
@@ -28,7 +28,6 @@ import (
        "github.com/containerd/containerd/mount"
        "github.com/containerd/continuity/fs"
        "github.com/containerd/continuity/testutil/loopback"
-       "golang.org/x/sys/unix"
 )

 func setupSnapshotter(t *testing.T) ([]Opt, error) {
@@ -51,8 +50,6 @@ func setupSnapshotter(t *testing.T) ([]Opt, error) {
        if out, err := exec.Command(mkfs, loop.Device).CombinedOutput(); err != nil {
                return nil, fmt.Errorf("failed to make ext4 filesystem (out: %q): %w", out, err)
        }
-       // sync after a mkfs on the loopback before trying to mount the device
-       unix.Sync()

        if err := testMount(t, loop.Device); err != nil {
                return nil, err
@@ -84,5 +81,5 @@ func testMount(t *testing.T, device string) error {
        if err := os.Remove(filepath.Join(root, "lost+found")); err != nil {
                return err
        }
-       return mount.UnmountAll(root, unix.MNT_DETACH)
+       return mount.UnmountAll(root, 0)
 }
diff --git a/snapshots/testsuite/testsuite.go b/snapshots/testsuite/testsuite.go
index 7afbd953e..36982adab 100644
--- a/snapshots/testsuite/testsuite.go
+++ b/snapshots/testsuite/testsuite.go
@@ -959,7 +959,6 @@ func check128LayersMount(name string) func(ctx context.Context, t *testing.T, sn
                                t.Fatalf("[layer %d] preparing doesn't equal to flat after apply: %+v", i, err)
                        }

-                       sync()
                        testutil.Unmount(t, preparing)

                        parent = filepath.Join(work, fmt.Sprintf("committed-%d", i))
```

no idea what happened to this testsuite.

Reply:
Just on the dev box?

What happens if we remove the `sync` option on mount as well?

Reply:
My devbox has bad IO performance. I can't reproduce the CI issue or the issue mentioned in the previous reply in my homelab device which has better disk device :joy:.

After `remove `sync` option, it shows

```
testsuite.go:944: [layer 100] preparing doesn't equal to flat before apply: directory diff between /tmp/snapshot-suite-Blockfile-2913520526/work/flat and /tmp/snapshot-suite-Blockfile-2913520526/work/prepare-layer-100
            + /onlyme/file-98
            - /addhere/file-99
            - /onlyme/file-99
            ~ /overwriteme(mode: 777, uid: 0, gid: 0) -> /overwriteme(mode: 777, uid: 0, gid: 0)

```